searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/UnzipFile.java

```diff
@@ -1,19 +1,27 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * Licence agreement you entered into with Jalasoft
+ */
 package org.jalau.at18.searchobject.model.objectrecognizer;
 import org.jalau.at18.searchobject.common.exception.UnzipFileException;
 
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
-import java.io.IOException;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipInputStream;
 import java.nio.file.Path;
 public class UnzipFile {
     Path dir;
-    public UnzipFile(Path path) throws IOException {
+    public UnzipFile(Path path) throws UnzipFileException {
         unzip(path);
     }
-    private void unzip(Path path) throws IOException {
+    private void unzip(Path path) throws UnzipFileException {
+        try {
             Path fileZip = path;
```

Comment:
this line is not necesary, and we can have line 26 like `File destDir = new File(getFolder(path.toString()));`